src/internal/m365/service/groups/restore.go

```diff
@@ -71,16 +66,45 @@ func ConsumeRestoreCollections(
 
 		switch dc.FullPath().Category() {
 		case path.LibrariesCategory:
-			// TODO(meain): As of now we only restore the root site
-			// and that too to whatever is currently the root site of the
-			// group and not the original one. Not sure if the
-			// original can be changed.
-			resp, err = ac.Groups().GetRootSite(ctx, rcc.ProtectedResource.ID())
-			if err != nil {
-				return nil, err
+			siteID := dc.FullPath().Folders()[1]
+
+			webURL, ok := backupSiteIDWebURL.NameOf(siteID)
+			if !ok {
+				// This should not happen, but just in case
+				logger.Ctx(ctx).With("site_id", siteID).Info("site weburl not found, using site id")
+			}
+
+			// In case a site that we had previously backed up was
+			// deleted, skip that site with a warning.
+			siteName, ok := siteNames[webURL]
+			if !ok {
+				site, err := ac.Sites().GetByID(ctx, webURL, api.CallConfig{})
+				if err != nil {
+					siteNames[webURL] = ""
+
+					if graph.IsErrSiteCouldNotBeFound(err) {
+						// TODO(meain): Should we surface this to the user somehow?
```

Comment:
How should we surface this information to the user without failing the backup?

Reply:
Created a tech-debt ticket for this. Lemme follow up on this after this.
https://github.com/alcionai/corso/issues/4418